util/codegen/codegen_test.go

```diff
@@ -174,3 +176,88 @@ func TestGenericContainsPointers(t *testing.T) {
 		})
 	}
 }
+
+func TestAssertStructUnchanged(t *testing.T) {
+	type args struct {
+		t      *types.Struct
+		tname  string
+		params *types.TypeParamList
+		ctx    string
+		it     *ImportTracker
+	}
+
+	// package t1 with a struct T1 with two fields
+	p1 := types.NewPackage("t1", "t1")
+	p1.Scope().Insert(types.NewTypeName(0, p1, "T1", types.NewStruct([]*types.Var{
+		types.NewField(0, nil, "P1", types.Typ[types.Int], false),
+		types.NewField(0, nil, "P2", types.Typ[types.String], false),
+	}, nil)))
+
+	// package t2 with a struct T2 with a named field T1 T1 and two other fields
+	p2 := types.NewPackage("t2", "t2")
+	p2.Scope().Insert(types.NewTypeName(0, p2, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil)))
+	p2.Scope().Insert(types.NewTypeName(0, p2, "T2", types.NewStruct([]*types.Var{
+		types.NewField(0, nil, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil), false),
+		types.NewField(0, nil, "P1", types.Typ[types.Int], false),
+		types.NewField(0, nil, "P2", types.Typ[types.String], false),
+	}, nil)))
+
+	// package t3 with a struct T3 with an embedded field T1 and two other fields
+	p3 := types.NewPackage("t3", "t3")
+	p3.Scope().Insert(types.NewTypeName(0, p3, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil)))
+	p3.Scope().Insert(types.NewTypeName(0, p3, "T2", types.NewStruct([]*types.Var{
+		types.NewField(0, nil, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil), true),
+		types.NewField(0, nil, "P1", types.Typ[types.Int], false),
+		types.NewField(0, nil, "P2", types.Typ[types.String], false),
+	}, nil)))
+
+	tests := []struct {
+		name string
+		args args
+		want []byte
+	}{
+		{
+			name: "t1-internally_defined",
+			args: args{
+				t:      p1.Scope().Lookup("T1").Type().Underlying().(*types.Struct),
+				tname:  "prefix_",
+				params: nil,
+				ctx:    "",
+				it:     NewImportTracker(p1),
+			},
+			want: []byte("var _prefix_NeedsRegeneration = prefix_(struct {\n\tP1 int \n\tP2 string \n}{})"),
+		},
+		{
+			name: "t2-with_named_field",
+			args: args{
+				t:      p2.Scope().Lookup("T2").Type().Underlying().(*types.Struct),
```

Comment:
optional: define `T2` here? There's probably no reason to insert it into `p2`'s scope on line 199 just to look it up here. Same for `T3`.

Reply:
You are right, no need to define package previously.